pkg/util/mount/mount_linux_test.go

```diff
@@ -1179,6 +1180,25 @@ func TestBindSubPath(t *testing.T) {
 			},
 			expectError: false,
 		},
+		{
+			name: "subpath-mounting-unix-socket",
```

Comment:
Also a testcase where socket is not the last segment in the path.

Reply:
Also testcase for fifo?

Reply:
added tests. also added test for fifo